src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -1205,6 +1214,52 @@ class NPlusOneDBSpanDetectorExtended(NPlusOneDBSpanDetector):
     )
 
 
+class FileIOMainThreadDetector(PerformanceDetector):
+    """
+    Checks for a file io span on the main thread
+    """
+
+    __slots__ = ("spans_involved", "stored_problems")
+
+    settings_key = DetectorType.FILE_IO_MAIN_THREAD
+
+    def init(self):
+        self.spans_involved = {}
+        self.most_recent_start_time = {}
+        self.most_recent_hash = {}
+        self.stored_problems = {}
+
+    def visit_span(self, span: Span):
+        if self._is_file_io_on_main_thread(span):
+            settings_for_span = self.settings_for_span(span)
+            if not settings_for_span:
+                return
+
+            op, span_id, op_prefix, span_duration, settings = settings_for_span
+            if span_duration.total_seconds() * 1000 > settings["duration_threshold"]:
+                call_stack = ".".join(
+                    [item["function"] for item in span.get("data", {}).get("call_stack", [])]
+                ).encode("utf8")
+                hashed_stack = hashlib.sha1(call_stack).hexdigest()
+                fingerprint = f"1-{GroupType.PERFORMANCE_FILE_IO_MAIN_THREAD}-{hashed_stack}"
+                self.stored_problems[fingerprint] = PerformanceProblem(
+                    fingerprint=fingerprint,
+                    op=span.get("op"),
+                    desc=span.get("description", ""),
+                    parent_span_ids=[],
+                    type=GroupType.PERFORMANCE_FILE_IO_MAIN_THREAD,
+                    cause_span_ids=[span.get("span_id", None)],
+                    offender_span_ids=[span.get("span_id", None)],
+                )
+
+    def _is_file_io_on_main_thread(self, span: Span) -> bool:
+        data = span.get("data", {})
+        if data is None:
+            return False
+        # doing is True since the value can be any type
+        return data.get("blocked_ui_thread", False) is True
```

Comment:
Should there be any restrictions on the span op here as well, or will all spans with `blocked_ui_thread` always be file I/O? Based on the name I could see it getting used on other operations down the line too.

Reply:
Hmm I'll add a prefix check for `file` good callout :+1:

Reply:
In the rfc https://github.com/getsentry/rfcs/pull/36 we say that this flag works only for auto-instrumented spans, and when we know for sure that the span spent its entire time on the main/ui thread. We also mentioned that we can potentially expand it to db and http ops.

So I'm not sure if we should make the detector generic already now, so we can add new perf issues without any changes here, or maintain the list of allowed ops whenever we decide to add something new

Reply:
Changing the code here in the detector is pretty easy. But I'll remove the op check for now. We aren't actually creating the issues yet so I'd rather collect too many metrics than too few